include/zephyr/drivers/video.h

```diff
@@ -318,6 +408,84 @@ static inline int video_get_format(const struct device *dev,
 	return api->get_format(dev, ep, fmt);
 }
 
+/**
+ * @brief Set video frame interval.
+ *
+ * Configure video device with a specific frame interval.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param frmival Pointer to a video frame interval struct.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOSYS If API is not implemented.
+ * @retval -EINVAL If parameters are invalid.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_set_frmival(const struct device *dev, enum video_endpoint_id ep,
+				    struct video_frmival *frmival)
+{
+	const struct video_driver_api *api = (const struct video_driver_api *)dev->api;
+
+	if (api->set_frmival == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->set_frmival(dev, ep, frmival);
+}
+
+/**
+ * @brief Get video frame interval.
+ *
+ * Get current frame interval of the video device.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param ep Endpoint ID.
+ * @param frmival Pointer to a video frame interval struct.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOSYS If API is not implemented.
+ * @retval -EINVAL If parameters are invalid.
+ * @retval -EIO General input / output error.
+ */
+static inline int video_get_frmival(const struct device *dev, enum video_endpoint_id ep,
+				    struct video_frmival *frmival)
+{
+	const struct video_driver_api *api = (const struct video_driver_api *)dev->api;
+
+	if (api->get_frmival == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->get_frmival(dev, ep, frmival);
+}
+
+/**
+ * @brief List video frame intervals.
+ *
+ * List all supported video frame intervals of a given format.
```

Comment:
I like this API as it allows to lookup for other ~~`sensor`~~ ~~`sdev`~~ ~~`source_dev`~~ `source` devices to see what they support, although it is different from [`video_get_caps()`](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/video.h#L466-L468). Just observing.

(reviewers: see [line 103](https://github.com/zephyrproject-rtos/zephyr/pull/72254/files#diff-5c275ef65ca2e3a0087625307d2de00a3c1edaa2d9edfa416ac4993a1755ffc5R103))

Would it help to comment how one is expected to walk through the enumeration? Stating that "`frmival` acts as a query" would also have helped with novice users (like me) to understand [this](https://github.com/zephyrproject-rtos/zephyr/pull/72254/files#r1590003732).

Reply:
ok, I added some texts